pilot/pkg/xds/discovery.go

```diff
@@ -363,6 +363,15 @@ func (s *DiscoveryServer) handleUpdates(stopCh <-chan struct{}) {
 	debounce(s.pushChannel, stopCh, s.debounceOptions, s.Push, s.CommittedUpdates)
 }
 
+// extractFirstConfigKey grabs the "first" element in the map. This is random, so this function
```

Comment:
Map ordering is random like you said, but here is an alternative that might be more explicit (up to you):
```go
keys := reflect.ValueOf(c).MapKeys()
return keys[rand.Intn(len(keys))].Interface()
```